jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/util/DirectDataCloudConnection.java

```diff
@@ -47,11 +101,259 @@ public static DataCloudConnection of(String url, Properties properties) throws S
+            throw new DataCloudJDBCException("Mutual TLS context creation failed: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Validates that truststore file exists and is readable.
+     */
+    private static void validateTruststoreFile(String truststorePath) throws DataCloudJDBCException {
+        File truststoreFile = new File(truststorePath);
+        if (!truststoreFile.exists()) {
+            throw new DataCloudJDBCException("Truststore file not found: " + truststorePath
+                    + ". Ensure the JKS truststore exists and the path is correct.");
+        }
+        if (!truststoreFile.canRead()) {
+            throw new DataCloudJDBCException(
+                    "Truststore file is not readable: " + truststorePath + ". Check file permissions.");
+        }
+        if (truststoreFile.length() == 0) {
+            throw new DataCloudJDBCException("Truststore file is empty: " + truststorePath);
+        }
+    }
+
+    /**
+     * Validates that all certificate files for mutual TLS exist and are readable.
+     */
+    private static void validateCertificateFiles(String clientCertPath, String clientKeyPath, String caCertPath)
+            throws DataCloudJDBCException {
+        validatePemFile(clientCertPath, "Client certificate");
+        validatePemFile(clientKeyPath, "Client private key");
+        validatePemFile(caCertPath, "CA certificate");
+
+        // Additional validation - ensure client cert and key are a matching pair
+        // This is a basic check - more sophisticated validation could be added
+        log.debug("Certificate file validation completed successfully");
+    }
+
+    /**
+     * Validates a PEM certificate file.
+     */
+    private static void validatePemFile(String path, String description) throws DataCloudJDBCException {
+        File file = new File(path);
+        if (!file.exists()) {
+            throw new DataCloudJDBCException(
+                    description + " file not found: " + path + ". Ensure the PEM file exists and the path is correct.");
+        }
+        if (!file.canRead()) {
+            throw new DataCloudJDBCException(
+                    description + " file is not readable: " + path + ". Check file permissions.");
+        }
+        if (file.length() == 0) {
+            throw new DataCloudJDBCException(description + " file is empty: " + path);
+        }
+
+        // Basic PEM format validation - check for PEM markers (Java 8 compatible)
+        try (java.util.Scanner scanner = new java.util.Scanner(file)) {
+            String content = scanner.useDelimiter("\\A").next();
+            if (!content.contains("-----BEGIN") || !content.contains("-----END")) {
+                log.warn("{} file may not be in valid PEM format: {}", description, path);
+            }
+        } catch (Exception e) {
+            log.warn("Could not validate PEM format for {}: {}", description, e.getMessage());
+        }
```

Comment:
I would pull this out into a private method named `basicValidationForPEM` or something, just informed by the comment